internal/middleware/logging.go

```diff
@@ -2,25 +2,56 @@ package middleware
 
 import (
 	"context"
+	"encoding/json"
 	"time"
 
+	"github.com/google/uuid"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"github.com/oklog/ulid/v2"
 	"github.com/openfga/openfga/pkg/logger"
 	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 )
 
 func NewLoggingInterceptor(logger logger.Logger) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		start := time.Now()
 
-		resp, err := handler(ctx, req)
+		id, err := uuid.NewRandom()
+		if err != nil {
+			logger.Error("failed to generate uuid", zap.Error(err))
+		}
+		requestID := id.String()
+
+		err = grpc.SetHeader(ctx, metadata.Pairs("X-Request-Id", requestID))
```

Comment:
Can we move this into it's own middleware `internal/middleware/requestid`?

Inject the autogenerated request id into the context and extract it in this middleware (if it's present) and add it to the logged fields.

This way we're separating concerns. Having the request id in the context is beneficial beyond just having it for logging, so this is why I believe it warrants separation of concerns.

Reply:
Good suggestion! Thanks! Done in #478. It will need to be reviewed first.